prover/src/constraints/commitment.rs

```diff
@@ -18,44 +19,44 @@ use super::RowMatrix;
 ///
 /// The commitment consists of two components:
 /// * Evaluations of composition polynomial columns over the LDE domain.
-/// * Merkle tree where each leaf in the tree corresponds to a row in the composition polynomial
-///   evaluation matrix.
-pub struct ConstraintCommitment<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>> {
+/// * Vector commitment where each vector element corresponds to the digest of a row in
+///   the composition polynomial evaluation matrix.
+pub struct ConstraintCommitment<
+    E: FieldElement,
+    H: ElementHasher<BaseField = E::BaseField>,
+    V: VectorCommitment<H>,
+> {
     evaluations: RowMatrix<E>,
-    commitment: MerkleTree<H>,
+    vector_commitment: V,
+    _h: PhantomData<H>,
 }
 
-impl<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>> ConstraintCommitment<E, H> {
+impl<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>, V: VectorCommitment<H>>
+    ConstraintCommitment<E, H, V>
+{
     /// Creates a new constraint evaluation commitment from the provided composition polynomial
-    /// evaluations and the corresponding Merkle tree commitment.
-    pub fn new(evaluations: RowMatrix<E>, commitment: MerkleTree<H>) -> ConstraintCommitment<E, H> {
-        assert_eq!(
-            evaluations.num_rows(),
-            commitment.leaves().len(),
-            "number of rows in constraint evaluation matrix must be the same as number of leaves in constraint commitment"
-        );
```

Comment:
Similar to one of the previous comments: if we had `VectorCommitment::domain_size(&self) -> usize`, I think we'd be able to bring back this assert.

Reply:
added back